python/mxnet/lr_scheduler.py

```diff
@@ -69,16 +76,45 @@ def __call__(self, num_update):
             the maximal number of updates applied to a weight.
         """
 
-        # NOTE: use while rather than if  (for continuing training via load_epoch)
-        while num_update > self.count + self.step:
+        # recover learning rate for continuing training via load_epoch.
+        if not self.lr_recovered:
```

Comment:
It is used to indicated whether the learning rate is recovered. For a normal training, it is simply set to True when the function is first called. For a continual training, it recovers to the correct base_lr. It is actually an indicator that this code block https://github.com/taoari/mxnet/blob/8c270ca2372e39f6e64ea4564c9b91e66d0b5b6f/python/mxnet/lr_scheduler.py#L79_L94 is only executed when the function is first called.